### PR TITLE
feat(core): add welcome route

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add welcome route:

1. Rename `koaProxyGuard` to `koaSpaSessionGuard` that only guard spaProxy. 
2. Exrtact `koaRootProxy` throw 404 for now
3. Add `koaCheckAdmin` middleware for /welsome path. 
4. Add query hasAdminUser

## Linear Issue Reference

log-2882

## Testing

ut case added
test locally
@logto-io/eng

### CODE REVIEW

File: packages/core/src/middleware/koa-check-admin.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import { hasAdminUsers } from '@/queries/user';
+
+export default function koaWelcomeProxy<
+  StateT,
+  ContextT extends IRouterParamContext,
+  ResponseBodyT
+>(): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  return async (ctx) => {
+    if (await hasAdminUsers()) {
```
Comment (demonzoo) [inline packages/core/src/middleware/koa-check-admin.ts:12]: Hmmm, I'm just thinking, what if a user registered an admin account, and revoke the admin role? Then there would be a non-admin user in the DB, but `hasAdminUsers` will return false

Comment (demonzoo) [reply packages/core/src/middleware/koa-check-admin.ts:12]: I think there should be a rule that one cannot change the role himself. Or even there should be a super admin (owner) of the tenant, whose role cannot be changed, or can only be passed to another user, like Slack workspace owner.

File: packages/core/src/middleware/koa-root-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  ResponseBodyT
+>(): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  return async (ctx, next) => {
+    const requestPath = ctx.request.path;
+
+    // Empty path return 404
+    if (requestPath === '/') {
+      ctx.throw(404);
+
+      return;
```
Comment (demonzoo) [inline packages/core/src/middleware/koa-root-proxy.ts:16]: Maybe we can redirect to `/console`?

**Review by demonzoo — APPROVED** [review]
Overall looks good, but there might be some flaws in the `hasAdminUser` part. Please check.

File: packages/core/src/middleware/koa-check-admin.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/middleware/koa-check-admin.ts:12]: Agree, what @demonzoo said can be a potential security issue.

File: packages/core/src/middleware/koa-root-proxy.ts (hunk 2 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-root-proxy.ts:16]: There is another short discussion in the previous PR. Keep 404 for now.

File: packages/core/src/middleware/koa-check-admin.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-check-admin.ts:12]: I might prefer to let the role control flow handle that.  Under current status, if all admin-account has been revoked, technically we should enable the register flow. 
 In another word, as you mentioned,  I think revoking all admin role action itself puts the whole system into an unstable status. Not checking `admin roles` and allowing register logic.

Comment (simeng-li) [reply packages/core/src/middleware/koa-check-admin.ts:12]: Unless we put a more strict rule on this, only 0 users found, could one create a new admin account.

File: packages/core/src/middleware/koa-root-proxy.ts (hunk 2 above)
Comment (demonzoo) [reply packages/core/src/middleware/koa-root-proxy.ts:16]: ok

File: packages/core/src/middleware/koa-check-admin.ts (hunk 1 above)
Comment (demonzoo) [reply packages/core/src/middleware/koa-check-admin.ts:12]: I think this deserves a discussion. Mind creating a ticket and bringing it to the team meeting real quick?

**Review by wangsijie — APPROVED** [review]

**Pull request merged**